packages/backend/src/open_payments/account/service.ts

```diff
@@ -69,3 +71,82 @@ async function getAccount(
 ): Promise<Account | undefined> {
   return await Account.query(deps.knex).findById(id).withGraphJoined('asset')
 }
+
+// Fetch (and lock) an account for work.
+// Returns the id of the processed account (if any).
+async function processNextAccount(
+  deps_: ServiceDependencies
+): Promise<string | undefined> {
+  return deps_.knex.transaction(async (trx) => {
+    const now = new Date(Date.now()).toISOString()
+    const accounts = await Account.query(trx)
+      .limit(1)
+      // Ensure the accounts cannot be processed concurrently by multiple workers.
+      .forUpdate()
+      // If an account is locked, don't wait — just come back for it later.
+      .skipLocked()
+      .where('processAt', '<=', now)
+      .withGraphFetched('asset')
+
+    const account = accounts[0]
+    if (!account) return
+
+    const deps = {
+      ...deps_,
+      knex: trx,
+      logger: deps_.logger.child({
+        account: account.id
+      })
+    }
+
+    await createWithdrawalEvent(deps, account)
+
+    await account.$query(deps.knex).patch({
+      processAt: null
+    })
+
+    return account.id
+  })
+}
+
+async function createWithdrawalEvent(
```

Comment:
Add a comment noting that for this to be safe, the provided `account` *must* be fetched with `deps.knex`'s transaction.